internal/ethapi/transaction_args.go

```diff
@@ -135,6 +135,12 @@ func (args *TransactionArgs) setDefaults(ctx context.Context, b Backend) error {
 			return errors.New(`contract creation without any data provided`)
 		}
 	}
+	// Assign a very high gas limit for cases where an accurate gas limit is not critical,
```

Comment:
Is this needed? I think a max value will be assigned in `args.ToMessage`.

Reply:
Hm but no default is set in `args.toTransaction`. So I guess its fine

Reply:
Yeah. But `setDefaults` is called before `args.ToMessage` in some functions, if we want to bypass `DoEstimateGas` to gain some performance, then need to add a bypass param in `setDefaults`.